IdentityCore/src/util/mac/MSIDAppExtensionUtil.m

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+
+#import "MSIDAppExtensionUtil.h"
+#import "MSIDMainThreadUtil.h"
+
+@implementation MSIDAppExtensionUtil
+
++ (BOOL)isExecutingInAppExtension
+{
+    NSString *mainBundlePath = [[NSBundle mainBundle] bundlePath];
+    
+    if (mainBundlePath.length == 0)
```

Comment:
Please thing to reuse ```msidIsStringNilOrBlank``` from ```MSIDExtensions```

Reply:
I created this class as a copy of existing src/util/ios/MSIDAppExtensionUtil.m with modifications for mac. The ios one seemed to use this check so I left the existing code as is. I'm not sure if the bundlePath can be blank but I'll make changes to use msidIsStringNilOrBlank for consistency.

Reply:
Addressed